cpp/src/arrow/compute/kernels/api.h

```diff
@@ -0,0 +1,29 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+// NOTE: API is EXPERIMENTAL and will change without going through a
+// deprecation cycle
+
+#pragma once
+
+/// \defgroup compute-kernels Extra compute kernels
+/// @{
+/// @}
+
+#include "arrow/compute/kernels/registry.h"  // IWYU pragma: export
+
+static arrow::Status g_kernels_registered = arrow::compute::RegisterComputeKernels();
```

Comment:
Can we do this in `.cc` in `cpp/src/arrow/compute/`?
It may fix the Ruby error and https://github.com/apache/arrow/pull/45618/files#diff-aae6568901bc7e5294f3998b11c62d24bf707b6dde25b18bf7f37a2e640e4884R92-R93 .

Reply:
Do you mean to create `cpp/src/arrow/compute/kernels/api.cc` or on `cpp/src/arrow/compute/kernels/registry.cc` or on a different file on `cpp/src/arrow/compute`?
I did try adding it on `cpp/src/arrow/compute/kernels/registry.cc` but the compiler seems to optimize it out as I wasn't using any code from it and it wasn't registering the kernels on DLL linkage.
I can try different approaches but I wasn't able to make it work with others. I can try to create an `api.cc` which is the only thing I haven't tried.

Reply:
Yes.

I'll try it on local later.

Reply:
I've tried a couple of things but I can't seem to make it work adding the static initializer on the code file instead of the header. An example gist of the diff I was trying around can be found here: https://gist.github.com/raulcd/a5fff3076212f11d5bb43a9c3e2814b4